rfcs/plugin-api/experiment-2.ts

```diff
@@ -0,0 +1,48 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+/* eslint-disable @typescript-eslint/no-empty-function */
+import { Config, EnvironmentFunctions } from "@inlang/core/src/config"
+
+export const myPlugin = createPlugin<{ pathPattern: string }>(({ pluginConfig, env }) => {
+	return {
+		id: "samuelstroschein.plugin-json",
+		defineConfig: (config) => {
+			if (pluginConfig.pathPattern === undefined) {
+				throw new Error("pathPattern is required")
+			}
+			config.readResources = readResources({ pluginConfig, env })
+			config.languages = getLanguages({ pluginConfig, env })
+		},
+	}
+})
+
+/**
+ * The function to configure a plugin.
+ *
+ * @example
+ *   plugins: [
+ * 	 	myPlugin({
+ * 	   	pathPattern: "hello",
+ * 	 	})
+ *   ]
+ */
+type PluginConfigFunction<PluginConfig> = (
+	pluginConfig: PluginConfig,
+) => (env: EnvironmentFunctions) => Plugin
+
+type Plugin = {
+	id: string
+	defineConfig(config: Partial<Config>): void
```

Comment:
this must return a `MaybePromise`! Or else we won't be able to add the json plugin in the SDK plugin as a fallback if missing

Reply:
Yes, in my mind I was already about to change this. Thank's for pinging!